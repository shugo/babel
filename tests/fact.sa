class MAIN is
   fact(n: INT): INT is
      if n = 0 then
         return 1;
      else
         return n * fact(n - 1);
      end;
   end;

   main is
      #OUT + "fact(10)=" + fact(10) + "\n";
   end;
end
