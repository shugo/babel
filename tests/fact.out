fact(10)=3628800
